src/OpenTelemetry.Extensions/Trace/RateLimitingSampler.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Based on the jaeger remote sampler for Java from https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/jaeger-remote-sampler/src/main/java/io/opentelemetry/sdk/extension/trace/jaeger/sampler/RateLimitingSampler.java
+
+using System.Globalization;
+using OpenTelemetry.Extensions.Internal;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Rate limiting sampler that can be used to sample traces at a constant rate.
+/// </summary>
+public class RateLimitingSampler : Sampler
+{
+    private const string SAMPLERTYPE = "ratelimiting";
+    private const string SAMPLERTYPEKEY = "sampler.type";
+    private const string SAMPLERPARAMKEY = "sampler.param";
+
+    private readonly RateLimiter rateLimiter;
+    private readonly SamplingResult onSamplingResult;
+    private readonly SamplingResult offSamplingResult;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="RateLimitingSampler"/> class.
+    /// </summary>
+    /// <param name="maxTracesPerSecond">The maximum number of traces that will be emitted each second.</param>
+    public RateLimitingSampler(int maxTracesPerSecond)
+    {
+        double maxBalance = maxTracesPerSecond < 1.0 ? 1.0 : maxTracesPerSecond;
+        this.rateLimiter = new RateLimiter(maxTracesPerSecond, maxBalance);
+        var attributes = new Dictionary<string, object>()
+        {
+            { SAMPLERTYPEKEY, SAMPLERTYPE },
+            { SAMPLERPARAMKEY, (double)maxTracesPerSecond },
```

Comment:
Is there a reason why you're converting from int to double here?

Reply:
It's what the original Java does :-)

The sampling algorithm supports a sample rate of a double, but the wrapper is exposing only an integer rate. It could change to a double, and I want this to be consistent across languages.